samples/Server/src/routes/addUser.ts

```diff
@@ -26,20 +27,29 @@ interface AddUserParam {
 router.post('/:threadId', async function (req, res, next) {
   const addUserParam: AddUserParam = req.body;
   const threadId = req.params['threadId'];
-  const moderatorCredential = threadIdToModeratorCredentialMap.get(threadId);
 
-  const chatClient = new ChatClient(getEndpoint(), moderatorCredential);
+  // create a user from the adminUserId and create a credential around that
+  const credential = new AzureCommunicationTokenCredential({
+    tokenRefresher: async () => (await getToken(createUser(), ['chat', 'voip'])).token,
+    refreshProactively: true
+  });
+
+  const chatClient = new ChatClient(getEndpoint(), credential);
   const chatThreadClient = await chatClient.getChatThreadClient(threadId);
 
-  await chatThreadClient.addParticipants({
-    participants: [
-      {
-        id: { communicationUserId: addUserParam.Id },
-        displayName: addUserParam.DisplayName
-      }
-    ]
-  });
-  res.sendStatus(201);
+  try {
+    await chatThreadClient.addParticipants({
+      participants: [
+        {
+          id: { communicationUserId: addUserParam.Id },
+          displayName: addUserParam.DisplayName
+        }
+      ]
+    });
+    res.sendStatus(201);
+  } catch (err) {
+    res.sendStatus(404);
```

Comment:
nit: `500` error?

Reply:
The thread was not found so 404? Its not really a server I give up error but rather someone is trying to add themselves to a thread that the adminUserId is not part of. I am open to 500 as well. 500 to me always sounds like completely unexpected on the developer and its a big catch all for the server threw an exception.

Reply:
Is `thread not found` the only reason this throws an error? If so 404 yeah makes sense but add a comment to mention we return 404 for that reason